kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt

```diff
@@ -1,72 +1,1065 @@
+          "Numbers bigger than itself" {
+            assertAll(nonMinNorMaxValueDoubles) {
+              it shouldNotMatchGreaterThanOrEqual it.slightlyGreater()
+              it shouldNotMatchGreaterThanOrEqual it.muchGreater()
+              
+            }
+          }
+          
+          "Positive Infinity" {
+            assertAll(nonMinNorMaxValueDoubles) {
+              it shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+          }
+          
+          "NaN" {
+            assertAll(nonMinNorMaxValueDoubles) {
+              it shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+        
+      }
+      
+      "The non-numeric double" - {
+        "NaN" - {
+          "Should not be greater than or equal to" - {
+            "Itself" {
+              NaN shouldNotMatchGreaterThanOrEqual NaN
+            }
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NaN shouldNotMatchGreaterThanOrEqual it
+              }
+            }
+            
+            "Positive Infinity" {
+              NaN shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "Negative Infinity" {
+              NaN shouldNotMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+          }
+        }
+        
+        "Positive Infinity" - {
+          "Should be greater than or equal to" - {
+            "Itself" {
+              POSITIVE_INFINITY shouldMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "Negative Infinity" {
+              POSITIVE_INFINITY shouldMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                POSITIVE_INFINITY shouldMatchGreaterThanOrEqual it
+              }
+            }
+          }
+          
+          "Should not be greater than or equal to" - {
+            "NaN" {
+              POSITIVE_INFINITY shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+        
+        "Negative Infinity" - {
+          "Should be greater than or equal to" - {
+            "Itself" {
+              NEGATIVE_INFINITY shouldMatchGreaterThanOrEqual NEGATIVE_INFINITY
+            }
+          }
+          
+          "Should not be greater than or equal to" - {
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual it
+              }
+            }
+            
+            "Positive Infinity" {
+              NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual POSITIVE_INFINITY
+            }
+            
+            "NaN" {
+              NEGATIVE_INFINITY shouldNotMatchGreaterThanOrEqual NaN
+            }
+          }
+        }
+      }
+    }
+    
+    
+  }
+  
+  
+  private fun Double.toleranceValue(): Double {
+    return 5 * ulp
```

Comment:
This is fine, but I would use `ulp, `2 * ulp`, and `3 * ulp` rather that 5, 10, and 100. The larger values don't catch any extra errors, and will cause more values to be rounded to infinity.